src/main/java/nus/climods/model/module/Module.java

```diff
@@ -108,6 +161,96 @@ public List<SemestersEnum> getSemesters() {
             .map(SemestersEnum::fromValue).collect(Collectors.toList());
     }
 
+    /**
+     * Checks if module is offered in a specified semester
+     *
+     * @param semester semester
+     * @return true if module is offered else false
+     */
+    public boolean availableInSemester(SemestersEnum semester) {
+        return getSemesters().stream().anyMatch(availableSemester -> availableSemester.equals(semester));
+    }
+
+    /**
+     * Returns the lesson types in module
+     *
+     * @param semester semester
+     * @return lesson types
+     */
+    public Set<LessonType> getLessonTypes(SemestersEnum semester) {
+        requireNonNull(lessonTypesMap);
+        assert lessonTypesMap.containsKey(semester);
+
+        return lessonTypesMap.get(semester);
+    }
+
+    /**
+     * Checks if module offers a lesson type
+     *
+     * @param lessonType lesson type
+     * @return true if module offers the lesson type else false
+     */
+    public boolean hasLessonType(LessonType lessonType) {
+        requireNonNull(lessonTypesMap);
+        return lessonTypesMap.values().stream().flatMap(Set::stream).collect(Collectors.toSet()).contains(lessonType);
+    }
+
+    /**
+     * Checks if module offers a lesson type
+     *
+     * @param lessonType lesson type
+     * @param semester   semester
+     * @return true if module offers the lesson type else false
+     */
+    public boolean hasLessonType(LessonType lessonType, SemestersEnum semester) {
+        requireNonNull(lessonTypesMap);
+        return lessonTypesMap.get(semester).contains(lessonType);
+    }
+
+    /**
+     * Returns the lessons types that can be selected
+     * <p>
+     * Only lessons that have more than one slot (categorized by the class number) is considered selectable
+     * </p>
+     *
+     * @param semester semester
+     * @return a set of lesson types
+     */
+    public Set<LessonType> getSelectableLessonTypes(SemestersEnum semester) {
+        requireNonNull(lessonMap);
+        assert lessonMap.containsKey(semester);
+
+        Predicate<Map.Entry<LessonType, ModuleLessonIdMap>> selectableLessonPredicate =
+            lessonEntry -> lessonEntry.getValue().size() > 1;
+
+        return lessonMap.get(semester).entrySet().stream().filter(selectableLessonPredicate).map(Map.Entry::getKey)
+            .collect(Collectors.toSet());
+    }
+
```

Comment:
Actually i was also thinking if we should have a method to return `lessonTypes` that are not selectable? Not sure if we need it here or can just handle inside `PickCommand`.

Reply:
oh yea sure i think its fine, if you need it when implementing `pick` just let me know i can add it

Reply:
I have added it in the latest commit for convenience